linked_list/LinkedList.java

```diff
@@ -0,0 +1,218 @@
+/*
+* Following is the code for Linked List
+* T defines the type of Linked List to be created
+*/
+
+import java.lang.IndexOutOfBoundsException;
+import java.util.NoSuchElementException;
+
+// Class to create a Node that contains data and pointer to next Node 
+class Node<T> {
+    T data;     // Data of type T
+    Node<T> next;     // Pointer to next Node
+
+    // Constructor to create a Node
+    public Node(T input) {  
+        this.data = input;
+        this.next = null;
+    }
+}
+
+// Class containing methods to be performed on Linked List
+class LinkedList<T> {
+    
+    private Node<T> head;     // Pointer to first node of the Linked List
+    private int sizeOfList = 0;     // Size of the Linked List
+    
+    // Constructor to create an empty Linked List
+    public LinkedList() { 
+        head = null;
+    }
+    
+    // Method to add an element to the start of the Linked List
+    public void addFront(T input) {
+        Node<T> newNode = new Node<T>(input);     
+        newNode.next = head;    
+        head = newNode;     
+        sizeOfList++;       
+    }
+    
+    // Method to add an element to the end of the Linked List
+    public void addLast (T input) {
+        if(head == null) {     
+            Node<T> newNode = new Node<T>(input);
+            head = newNode;
+            sizeOfList = 1;
```

Comment:
This condition makes the first node of the list so I've written 

> sizeOfList = 1

just to be sure